accepted/2020/platform-checks/platform-checks.md

```diff
@@ -0,0 +1,450 @@
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple=false, Inherited=true)]
+    public sealed class AddedInOSPlatformAttribute : OSPlatformAttribute
+    {
+        public PlatformSpecificAttribute(string osPlatform,
+                                         int major,
+                                         int minor,
+                                         int build,
+                                         int revision);
+    }
+
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple=false, Inherited=true)]
+    public sealed class RemovedInOSPlatformAttribute : OSPlatformAttribute
+    {
+        public PlatformSpecificAttribute(string osPlatform,
+                                         int major,
+                                         int minor,
+                                         int build,
+                                         int revision);
+    }
+
+    [AttributeUsage(AttributeTargets.Assembly |
+                    AttributeTargets.Class |
+                    AttributeTargets.Constructor |
+                    AttributeTargets.Event |
+                    AttributeTargets.Method |
+                    AttributeTargets.Module |
+                    AttributeTargets.Property |
+                    AttributeTargets.Struct,
+                    AllowMultiple=false, Inherited=true)]
+    public sealed class ObsoletedInOSPlatformAttribute : OSPlatformAttribute
+    {
+        public PlatformSpecificAttribute(string osPlatform,
+                                         int major,
+                                         int minor,
+                                         int build,
+                                         int revision);
+        public string Url { get; set; }
+    }
+}
+```
+
+### Platform context
+
+To determine the platform context of the call site the analyzer must consider
+the following two configurations:
+
+1. **Call-site**. Application of `AddedInOSPlatformAttribute` to the containing
+   member, type, module, or assembly.
+
+2. **Project** The project TFMs and `TargetPlatformMinVersion`. If the project
+   has an platform-specific TFM (such as `net5.0-ios13.0`) we consider it to be
+   on the specified platform. If `TargetPlatformMinVersion` is set, we assume
+   that version, otherwise we assume the version as specified by the TFM.
```

Comment:
They don't today because they are only having access to what is passed to the compiler. I have worked this around in the platform-compat analyzer by writing a temp file to obj which is then passed to the compiler as an *additional file* which allows you pass any data you want. My understanding is that this is generally solved by the integration into the SDK. Certain build properties (such as TFM) will be passed as key-value pairs into a dictionary that is available to all analyzers.